File: Assets/FinGameWorks/Scripts/Datas/Enums.cs
using System;

namespace FinGameWorks.Scripts.Datas
{
    public class Enums
    {
        public enum DetectedActivityType
        {
            IN_VEHICLE,
            ON_BICYCLE,
            ON_FOOT,
            RUNNING,
            STILL,
            TILTING,
            UNKNOWN,
            WALKING
        }
        
        public enum WeatherType
        {
            CLEAR,	//Clear weather condition.
            CLOUDY,	//	Cloudy weather condition.
            FOGGY,	//	Foggy weather condition.
            HAZY,	//	Hazy weather condition.
            ICY,	//	Icy weather condition.
            RAINY,	//	Rainy weather condition.
            SNOWY,	//	Snowy weather condition.
            STORMY,	//	Stormy weather condition.
            UNKNOWN,	//	Unknown weather condition.
            WINDY,	//	Windy weather condition.
        }

        public static float NoiseFrequencyByWeatherType(WeatherType weatherType)
        {
            switch (weatherType)
            {
                case WeatherType.CLEAR:
                    return 0;
                case WeatherType.CLOUDY:
                    return 0.05f;
                case WeatherType.FOGGY:
                    return -0.05f;
                case WeatherType.HAZY:
                    return 0.02f;
                case WeatherType.ICY:
                    return -0.15f;
                case WeatherType.RAINY:
                    return 0.3f;
                case WeatherType.SNOWY:
                    return 0.1f;
                case WeatherType.STORMY:
                    return 0.4f;
                case WeatherType.UNKNOWN:
                    return 0;
                case WeatherType.WINDY:
                    return 0.2f;
                default:
                    return 0;
            }
        }
        
        public static float SimulationSpeedAdditionByWeatherType(WeatherType weatherType)
        {
            switch (weatherType)
            {
                case WeatherType.CLEAR:
                    return 0;
                case WeatherType.CLOUDY:
                    return 0;
                case WeatherType.FOGGY:
                    return -0.10f;
                case WeatherType.HAZY:
                    return -0.05f;
                case WeatherType.ICY:
                    return -0.1f;
                case WeatherType.RAINY:
                    return 0.15f;
                case WeatherType.SNOWY:
                    return 0.05f;
                case WeatherType.STORMY:
                    return 0.2f;
                case WeatherType.UNKNOWN:
                    return 0;
                case WeatherType.WINDY:
                    return 0.15f;
                default:
                    return 0;
            }
        }
    }
}

File: Builds/Android/Vortex/launcher/src/main/java/com/justzht/vortex/data/Enums.java
package com.justzht.vortex.data;

import com.google.android.gms.awareness.state.Weather;
import com.google.android.gms.location.DetectedActivity;

public class Enums
{
    public enum DetectedActivityType
    {
        IN_VEHICLE,
        ON_BICYCLE,
        ON_FOOT,
        RUNNING,
        STILL,
        TILTING,
        UNKNOWN,
        WALKING
    }

    public static DetectedActivityType getDetectedActivityType(int type)
    {
        switch (type)
        {
            case DetectedActivity.IN_VEHICLE:
                return DetectedActivityType.IN_VEHICLE;
            case DetectedActivity.ON_BICYCLE:
                return DetectedActivityType.ON_BICYCLE;
            case DetectedActivity.ON_FOOT:
                return DetectedActivityType.ON_FOOT;
            case DetectedActivity.RUNNING:
                return DetectedActivityType.RUNNING;
            case DetectedActivity.STILL:
                return DetectedActivityType.STILL;
            case DetectedActivity.TILTING:
                return DetectedActivityType.TILTING;
            case DetectedActivity.UNKNOWN:
                return DetectedActivityType.UNKNOWN;
            case DetectedActivity.WALKING:
                return DetectedActivityType.WALKING;
            default:
                return DetectedActivityType.UNKNOWN;
        }
    }

    public enum WeatherType
    {
        CLEAR,	//Clear weather condition.
        CLOUDY,	//	Cloudy weather condition.
        FOGGY,	//	Foggy weather condition.
        HAZY,	//	Hazy weather condition.
        ICY,	//	Icy weather condition.
        RAINY,	//	Rainy weather condition.
        SNOWY,	//	Snowy weather condition.
        STORMY,	//	Stormy weather condition.
        UNKNOWN,	//	Unknown weather condition.
        WINDY,	//	Windy weather condition.
    }

    public static WeatherType getWeatherType(int value)
    {
        switch (value)
        {
            case Weather.CONDITION_CLEAR:return WeatherType.CLEAR;
            case Weather.CONDITION_CLOUDY:return WeatherType.CLOUDY;
            case Weather.CONDITION_FOGGY:return WeatherType.FOGGY;
            case Weather.CONDITION_HAZY:return WeatherType.HAZY;
            case Weather.CONDITION_ICY:return WeatherType.ICY;
            case Weather.CONDITION_RAINY:return WeatherType.RAINY;
            case Weather.CONDITION_SNOWY:return WeatherType.SNOWY;
            case Weather.CONDITION_STORMY:return WeatherType.STORMY;
            case Weather.CONDITION_UNKNOWN:return WeatherType.UNKNOWN;
            case Weather.CONDITION_WINDY:return WeatherType.WINDY;
            default:return WeatherType.UNKNOWN;
        }
    }
}
